CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/MainNavigator.kt

```diff
@@ -0,0 +1,33 @@
+package co.nimblehq.coroutine.ui.screens
+
+import androidx.fragment.app.Fragment
+import co.nimblehq.coroutine.R
+import co.nimblehq.coroutine.ui.base.*
+import co.nimblehq.coroutine.ui.screens.home.HomeFragmentDirections.Companion.actionHomeFragmentToSecondFragment
+import co.nimblehq.coroutine.ui.screens.second.SecondBundle
+import javax.inject.Inject
+
+interface MainNavigator : BaseNavigator
+
+class MainNavigatorImpl @Inject constructor(
+    fragment: Fragment
+) : BaseNavigatorImpl(fragment), MainNavigator {
+
+    override val navHostFragmentId = R.id.navHostFragment
+
+    override fun navigate(event: NavigationEvent) {
+        when (event) {
+            is NavigationEvent.Second -> navigateToSecond(event.bundle)
+        }
+    }
+
+    private fun navigateToSecond(bundle: SecondBundle) {
+        val navController = findNavController()
+        when (navController?.currentDestination?.id) {
+            R.id.homeFragment -> navController.navigate(
+                actionHomeFragmentToSecondFragment(bundle)
+            )
+            else -> unsupportedNavigation()
+        }
```

Comment:
Any test case for this?

Reply:
Could we have a test case for this, please? 😄

Reply:
@chornerman How about this, please?

Reply:
I think it's best to keep the test cases for a different PR, as we haven't integrated many of the testing dependencies yet in in this project 🙏

Reply:
@hoangnguyen92dn We have discussed about the test [here](https://github.com/nimblehq/android-templates/pull/68#discussion_r606902160), please let me know if you don't agree or have any other idea 🙏🏼